### PR TITLE
Feature/login divide functionality

### PR DESCRIPTION
**Pull request by tomklecha**
*added separate buttons for create and sign in user
*added neccesary tests
*added check currently logged user button

### CODE REVIEW

**Comment by tomklecha** [comment]
so i separate the class - but still something with the function is not working.
With or without result variable, not changing - still going straight to false. Even if after a while he sign in the user, or create a new one - depends which function you use

File: features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt [hunk 1]
```diff
@@ -74,6 +101,10 @@ internal class LoginPresenter(
         view.showError(message)
     }
 
+    private fun CoroutineScope.checkUser() = launch(dispatcher.IO) {
+        view.showError("Currently logged as: " + interactor.checkUser())
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt:105]: show error? what is the error here?

File: features/login/src/main/java/com/can_apps/login/core/LoginContract.kt [hunk 2]
```diff
@@ -23,27 +23,39 @@ internal interface LoginContract {
...
         fun onBackPressed()
 
-        fun onLoginClicked(password: String, loginName: String)
+        fun onSignClicked(password: String, loginName: String)
+
+        fun onCreateLoginClicked(password: String, loginName: String)
 
         fun logoutUser()
+
+        fun currentUser()
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginContract.kt:32]: what `currentUser` mean?

File: features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt (hunk 1 above)
Comment (Canato) [reply features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt:105]: view is on IO thread. Did you run this code?

Comment (Canato) [reply features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt:105]: What you do when checkUser return `null` ?

File: features/login/src/main/java/com/can_apps/login/core/LoginContract.kt [hunk 3]
```diff
@@ -23,27 +23,39 @@ internal interface LoginContract {
...
 
-        suspend fun loginUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
+        suspend fun signInUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
 
-        fun logoutUser()
+        suspend fun createUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
+
+        suspend fun logoutUser()
+
+        suspend fun checkUser(): String?
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginContract.kt:47]: We should return Domain. Or in edge case Boolean

File: features/login/src/main/java/com/can_apps/login/core/LoginContract.kt [hunk 4]
```diff
@@ -23,27 +23,39 @@ internal interface LoginContract {
...
 
-        suspend fun loginUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
+        suspend fun signInUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
 
-        fun logoutUser()
+        suspend fun createUser(loginName: LoginNameDomain, password: LoginPasswordDomain): LoginDomain
+
+        suspend fun logoutUser()
+
+        suspend fun checkUser() : String?
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginContract.kt:58]: We should return Domain. Or in edge case Boolean

File: features/login/src/main/java/com/can_apps/login/data/LoginFirebaseApi.kt [hunk 5]
```diff
@@ -7,18 +7,26 @@ internal interface LoginFirebaseApi {
...
+        return auth.signInWithEmailAndPassword(login, password).await().user == null
+    }
 
-    override suspend fun loginExistingUser(login: String, password: String): Boolean =
-        auth.signInWithEmailAndPassword(login, password).await().user == null
 
-    override fun logoutUser() {
+    override suspend fun checkUser(): String? {
+        return auth.currentUser?.email
+    }
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/data/LoginFirebaseApi.kt:27]: checkUser return user email? This name lead you to wrong information.

File: features/login/src/main/java/com/can_apps/login/data/LoginRepository.kt [hunk 6]
```diff
@@ -4,25 +4,43 @@ import com.can_apps.login.core.*
...
-    override suspend fun loginUser(
+    override suspend fun signInUser(
         loginName: LoginNameDomain,
         password: LoginPasswordDomain
     ): LoginDomain {
-        try {
+
+        return try {
             val result = api.loginExistingUser(loginName.value, password.value)
             if (!result) {
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/data/LoginRepository.kt:14]: why you need result variable?

File: features/login/src/main/java/com/can_apps/login/app/LoginFragment.kt [hunk 7]
```diff
@@ -32,17 +32,25 @@ class LoginFragment :
...
+            presenter.onCreateLoginClicked(
+                loginPasswordEditText.text.toString(),
+                loginNameEditText.text.toString()
+            )
         }
-    }
 
+        logoutButton.setOnClickListener { presenter.logoutUser() }
+
+        checkLoggedUser.setOnClickListener { presenter.currentUser() }
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/app/LoginFragment.kt:51]: signIn**Button**, createUser**Button**, logout**Button**, checkLoggedUser

We cannot know that checkLoggedUser is a button looking this code.

**Pull request merged**